ejb/src/main/java/org/wildfly/httpclient/ejb/HttpEJBInvocationBuilder.java

```diff
@@ -122,24 +124,28 @@ enum InvocationType {
 
     ClientRequest createRequest(final String prefix) {
         final ClientRequest clientRequest = new ClientRequest();
+        clientRequest.setMethod(getBeanRequestMethod());
         clientRequest.setPath(getBeanRequestPath(prefix));
         if (invocationType == InvocationType.METHOD_INVOCATION) {
-            clientRequest.setMethod(Methods.POST);
             clientRequest.getRequestHeaders().add(Headers.ACCEPT, INVOCATION_ACCEPT + "," + EJB_EXCEPTION);
             if (invocationId != null) {
                 clientRequest.getRequestHeaders().put(INVOCATION_ID, invocationId);
             }
             clientRequest.getRequestHeaders().put(Headers.CONTENT_TYPE, INVOCATION.toString());
         } else if (invocationType == InvocationType.STATEFUL_CREATE) {
-            clientRequest.setMethod(Methods.POST);
             clientRequest.getRequestHeaders().put(Headers.CONTENT_TYPE, SESSION_OPEN.toString());
             clientRequest.getRequestHeaders().add(Headers.ACCEPT, EJB_EXCEPTION.toString());
-        } else if(invocationType == InvocationType.CANCEL) {
-            clientRequest.setMethod(Methods.DELETE);
         }
         return clientRequest;
     }
 
+    private HttpString getBeanRequestMethod() {
```

Comment:
Ditto here. Avoid introducing an extra check for the invocation type that is unnecessary

Reply:
Same here, it is fixed later in this PR.